### DISCUSSION TITLE
Get list of machines: Weird error message

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#56] [Issue]
When trying to get the list of machines with

```python
requests.post(
    f"https://api.keygen.sh/v1/accounts/{account_id}/machines",
    headers={
        "Authorization": f"License {license_key}",
        "Content-Type": "application/vnd.api+json",
        "Accept": "application/vnd.api+json",
    },
).json()
```
I'm getting back
```python
{
    'meta': {'id': '55366052-6f97-476b-ae0b-6944f0d6d974'},
    'errors': [
        {
            'title': 'Bad request',
            'detail': 'is missing',
            'source': {'pointer': '/data'}
        }
    ]
}
```
Not sure what the issue is, but `is missing` is perhaps a bug in the error message generation.

**Comment by nschloe** [comment]
Whoops! Wrong request type, it should have been a `GET`. Not sure if anything can be done about the error message though.

**Comment by ezekg** [comment]
Glad you figured it out. The error message is saying that you're missing the `data` property in the request body, because it's required when creating a machine (i.e. when using the `POST` method on the `/machines` endpoint).
